accepted/0000-uyuni-microos-management.md

```diff
@@ -0,0 +1,201 @@
+- `util.mgr_switch_to_venv_minion`
+
+
+### Unsupported Internal States
+Not all internal states we have make sense on MicroOS / SUSE Linux Micro. These states are unsupported.
+
+- `appstreams.configure` - only useful for RHEL systems
+- `bootstrap.remove_traditional_stack` - traditional stack was never supported on these systems
+- `cocoattest.requestdata`- only supports SLES 15 SP6
+- `rebootifneeded` - The way this is written is incompatible with transactional systems
+- `uptodate` - The way this is written is incompatible with transactional systems
+
+### Configurable States (`java.salt_custom_states_use_transactional_update`)
+These states could either be applied with `state.apply` or `transactional_update.apply`. The Java
+backend code consults `java.salt_custom_states_use_transactional_update` to decide how to apply them.
+
+-   `custom`
+-   `custom_groups`
+-   `custom_org`
+-   `recurring`
+-   `remotecommands`
+
+### Required Changes
+- Extract `dmidecode` installation steps in `hardware.profileupdate` to `hardware.prereq`
+- Split `services.*` states and extract package installation steps
+
+### States not yet categorized
+- `bootstrap.autoinstall`
+- `channels.gpg-keys`
+- `scap` NOTE: `remediate=True` is likely OS-altering
+- `configuration.*` depends on path to the deployed/analyzed file
+
+## Automatic reboots during bootstrapping
+Uyuni bootstraps new systems through different mechanisms: `state.apply` over Salt SSH, running a
+bash script, or by reacting to a newly connected Salt Minion. During the bootstrap procedure, Uyuni
+installs the Salt Bundle on the client. This requires a reboot of transactional systems.
+
+### `state.apply` over Salt SSH
+This is the mechanism used when users navigate to "Systems > Bootstrapping" in the WebUI.
+
+Uyuni targets the new system with Salt SSH and always uses `state.apply` to apply the `bootstrap`
+state. This is required because Uyuni relies on information collected on the client system to know
+what kind of system it is. This includes finding out if the new system is a transactional system.
+The `bootstrap`state install our Salt Bundle package correctly on both traditionally-managed and
+transactional systems.
+
+Triggering a reboot over Salt SSH has one problem: rebooting before Salt finishes loses the job data
+of the `bootstrap` state. This can be avoided by systemd feature called "Inhibitor Lock", which can
+delay a shutdown request while Salt SSH runs.
+
+#### Add Inhibitor Lock to Salt SSH
+Applications can set _inhibitor locks_ to block or delay system shutdown and sleep states. Salt SSH
+will be changed to set a _delay_ inhibitor lock to stop the system from rebooting immediately. Salt
+SSH has time to return job results back to the Salt Master, unless it takes longer than
+_InhibitDelayMaxSecs_. This config setting is specified in `logind.conf(5)` and can't be overridden
+by Salt SSH. The default is 5 seconds. When Salt SSH requires more than 5 seconds to return the job
```

Comment:
I think it is a fair compromise. In theory reboot should be the last thing, so it should just be the time of retuning the event.